core/src/main/java/jenkins/security/BasicHeaderApiTokenAuthenticator.java

```diff
@@ -33,9 +34,20 @@ public Authentication authenticate(HttpServletRequest req, HttpServletResponse r
             Authentication auth;
             try {
                 UserDetails userDetails = u.getUserDetailsForImpersonation();
-                auth = u.impersonate(userDetails);
+                if (UserDetailsHelper.isFullyActive(userDetails)) {
+                    auth = u.impersonate(userDetails);
 
-                SecurityListener.fireAuthenticated(userDetails);
+                    SecurityListener.fireAuthenticated(userDetails);
+                } else {
+                    LOGGER.log(INFO, "API token matched for user {0} but the account is [enable={1}, accountNonExpired={2}, accountNonLocked={3}, credentialsNonExpired={4}]", new Object[]{
```

Comment:
```suggestion
                    LOGGER.log(INFO, "API token matched for user {0} but the account is [enabled={1}, accountNonExpired={2}, accountNonLocked={3}, credentialsNonExpired={4}]", new Object[]{
```

Reply:
resolved actually AFAICT

Reply:
This one was **not** commited, I was too fast when commiting the previous one and this one.